### PR TITLE
AEA-1647 libp2p_node: addresses announced mapping

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

libp2p_node: addresses announced mapping

## Fixes

https://github.com/fetchai/agents-aea/issues/2560

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go [hunk 1]
```diff
@@ -348,9 +349,10 @@ func New(opts ...Option) (*DHTPeer, error) {
 	// if peer is joining an existing network, announce my agent address if set
 	if len(dhtPeer.bootstrapPeers) > 0 {
 		// TOFIX, set to true even when no address present
-		dhtPeer.addressAnnounced = true
-		ldebug().Msg("Address was announced on bootstrap peers")
-		if dhtPeer.myAgentAddress != "" {
+		dhtPeer.enableAddressAnnouncement = true
```
Comment (DavidMinarsch) [inline libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go:352]: Can you explain?

File: libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go [hunk 2]
```diff
@@ -1433,72 +1434,67 @@ func (dhtPeer *DHTPeer) handleAeaNotifStream(stream network.Stream) {
...
-						Str("addr", addr).
-						Msg("while announcing delegate client address")
-				}
-			}
-			dhtPeer.tcpAddressesLock.RUnlock()
+		dhtPeer.tcpAddressesLock.RUnlock()
 
-		}
-		duration := timer.GetTimer(start)
-		opLatencyRegister.Observe(float64(duration.Microseconds()))
```
Comment (DavidMinarsch) [inline libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go:1498]: should we not keep these timers?

File: libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go (hunk 1 above)
Comment (solarw) [reply libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go:352]: during developement I found if no connections to other peers it's not possible to announce address, so initial idea of flag is check is possible to announce or not.

File: libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go (hunk 2 above)
Comment (solarw) [reply libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go:1498]: it's saved, just indent changes so github shows as removed

File: libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go (hunk 1 above)
Comment (DavidMinarsch) [reply libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go:352]: This comment could be left in code for posterity

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**